### PR TITLE
Add buffer distance into max_count and max_offset.

### PR DESCRIPTION
**Pull request by BG-AIMS**
Previously the buffering distance wasn't applied to max_count. Additionally have moved to max_offset to allow for full rotation of the search box without going outside of the rel_pix bounds. And adjusted options if there are no rel_pix for a search box (returns standard geometry to output valid DataFrame, but will be filtered out at later steps).

### CODE REVIEW

File: src/site_assessment/best_fit_polygons.jl [hunk 1]
```diff
@@ -35,6 +35,13 @@ positive.
 - Highest scoring rotation step
 - Highest scoring polygon
 - Quality control flag for site, indicating if `surr_threshold` was met in the highest scoring rotation.
+
+# Extended help
+The scores produced are a proportion of the polygon that are covered by valid pixel points,
+relative to the maximum number of points (`max_count`) (0-1). `max_count` is approximate
+(determined by user `x_dist` and `y_dist` box dimensions) and doesn't account for buffering
+and rotation of the search box. In rare cases scores could be > 1, however returned values
+are capped at max 1.
```
Comment (ConnectedSystems) [inline src/site_assessment/best_fit_polygons.jl:44]: As mentioned elsewhere, Extended help should directly below the main description text.

`# Returns` is typically the last section in a docstring

Comment (ConnectedSystems) [reply src/site_assessment/best_fit_polygons.jl:44]: As an aside, didn't we update the description for `rel_pix` in the argument list to at least indicate what `rel` means?

Comment (BG-AIMS) [reply src/site_assessment/best_fit_polygons.jl]: Addressed second comment in #39

**Comment by ConnectedSystems** [comment]
Merging into `streamlined-process` branch instead of `main`.

**Pull request merged**